packages/kbn-generate-csv/src/get_export_settings.ts

```diff
@@ -74,7 +78,21 @@ export const getExportSettings = async (
     timezone: setTimezone,
     scroll: {
       size: config.scroll.size,
-      duration: config.scroll.duration,
+      duration: ({ retryAt }, format = 'ms') => {
+        if (config.scroll.duration !== 'auto') {
+          return config.scroll.duration;
+        }
+
+        const now = new Date(Date.now()).getTime();
+        const timeTillRetry = new Date(retryAt!).getTime();
```

Comment:
Since we rely on `retryAt` from external code (Task Manager plugin), maybe we should throw an error if `retryAt` is null or undefined, and make sure that error will handled?